src/extension.ts

```diff
@@ -1731,9 +1736,34 @@ async function deleteContextKubernetes(explorerNode: explorer.KubernetesObject)
     }
 }
 
-async function useNamespaceKubernetes(explorerNode: explorer.KubernetesObject) {
-    if (await kubectlUtils.switchNamespace(kubectl, explorerNode.id)) {
-        refreshExplorer();
+async function useNamespaceKubernetes(explorerNode?: explorer.KubernetesObject) {
+    if (explorerNode) {
+        if (await kubectlUtils.switchNamespace(kubectl, explorerNode.id)) {
+            refreshExplorer();
+        }
+    } else {
+        promptKindName([kuberesources.allKinds.namespace], 'get', undefined, (resource) => {
+            if (resource) {
+                const currentNS = kubectlUtils.currentNamespace(kubectl);
+                currentNS.then((ns) => {
```

Comment:
Can you use `await` here?

Reply:
the arrow functions does not allow me to use await - mostly I guess I dont know :D . Can we use await in arrow function ?

Reply:
You should just need to make the arrow function async, e.g.

```
promptKindName(…, async (resource) => {
  const ns = await kubectlUtils.currentNamespace(…);  // should work now
));